Adapter_Revit_UI/AdapterActions/Remove.cs

```diff
@@ -0,0 +1,91 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+ 
+using Autodesk.Revit.DB;
+using Autodesk.Revit.UI;
+using BH.oM.Adapter;
+using BH.oM.Adapters.Revit;
+using BH.oM.Data.Requests;
+
+namespace BH.UI.Revit.Adapter
+{
+    public partial class RevitUIAdapter
+    {
+
+        /***************************************************/
+        /****             Protected Methods             ****/
+        /***************************************************/
+
+        public override int Remove(IRequest request, ActionConfig actionConfig = null)
+        {
+            // Check the document
+            UIDocument uiDocument = this.UIDocument;
+            Document document = this.Document;
+            if (document == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because Revit Document is null.");
+                return 0;
+            }
+
+            if (document.IsReadOnly)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because Revit Document is read only.");
+                return 0;
+            }
+
+            if (document.IsModifiable)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because another transaction is open in Revit.");
+                return 0;
+            }
+
+            // Set config
+            RevitRemoveConfig removeConfig = actionConfig as RevitRemoveConfig;
+            if (removeConfig == null)
+            {
+                BH.Engine.Reflection.Compute.RecordWarning("Revit Remove Config has not been specified. Default Revit Remove Config is used.");
+                removeConfig = RevitRemoveConfig.Default;
```

Comment:
This is what I meant: I'd prefer this line to become
`new RevitRemoveConfig()`

instead of having a `Default`. 

The only case I would see this as acceptable is if for some reason `RevitRemoveConfig.Default` has different default properties than a `new RevitRemoveConfig()`.

Reply:
I think I had a chat about it with @IsakNaslundBh at some point and decided to go for `public static readonly` property instead of empty constructor. I am not strict about that, would be good to hear others' opinion, @FraserGreenroyd @IsakNaslundBh @adecler ?

Reply:
Only reading the two comments from the pair of you, I personally agree with @alelom on this one I'm afraid - unless `RevitRemoveConfig` will do more than the default properties an initialisation would align with the rest of BHoM. But I have no strong opinion so am happy either way.

Reply:
@pawelbaran I thought your argument was about some speed issues, or not wanting a method on the class it self called default?

In general, I would agree with @alelom and @FraserGreenroyd , that a simple new is preferable, unless it means a massive slowdown (can not really see this being the case?) or any other massive issues that we not as deep in the code as you @pawelbaran , would know?

Reply:
No massive slowdown expected in this case for sure, happy to align having all 3 of you speaking with the same voice :)